runtime/gc_ctrl.c

```diff
@@ -171,6 +172,10 @@ CAMLprim value caml_gc_set(value v)
                      ARCH_INTNAT_PRINTF_FORMAT "u%%\n", caml_percent_free);
   }
 
+  newverb_gc = Long_val (Field (v, 3));
+
+  atomic_store_relaxed(&caml_verb_gc, newverb_gc);
```

Comment:
```suggestion
  atomic_store_relaxed(&caml_verb_gc, Long_val (Field (v, 3)));
```
and avoid the new variable declaration.